### PR TITLE
Reproduces #331

### PR DESCRIPTION
**Pull request by dpwspoon**

### CODE REVIEW

**Comment by dpwspoon** [comment]
@sanjay-saxena can you review this.  Note: the test is ignored because I think it is a bug.  Can you confirm.

File: management/src/test/java/org/kaazing/gateway/management/jmx/TlsTestUtil.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package org.kaazing.gateway.management.jmx;
+
+import java.io.FileInputStream;
+import java.security.KeyStore;
+
+public class TlsTestUtil {
+    static char[] password = "ab987c".toCharArray();
+    static String keyStoreFileLocation = "/Users/David/Desktop/kaazing-websocket-gateway-4.0.9/conf/keystore.db";
```
Comment (sanjay-saxena) [inline management/src/test/java/org/kaazing/gateway/management/jmx/TlsTestUtil.java]: How do we resolve`keyStoreFileLocation` absolute path when these tests are run in different environments?

File: management/src/test/java/org/kaazing/gateway/management/jmx/TlsTestUtil.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    static char[] password() {
+        return password;
```
Comment (sanjay-saxena) [inline management/src/test/java/org/kaazing/gateway/management/jmx/TlsTestUtil.java]: To be pedantic, can we just return a cloned password array?

File: management/pom.xml [hunk 3]
```diff
@@ -22,29 +22,29 @@
...
+            <dependency>
+                <groupId>org.kaazing</groupId>
+                <artifactId>snmp4j</artifactId>
+                <version>1.11.3</version>
+            </dependency>
+            <dependency>
+                <groupId>org.kaazing</groupId>
+                <artifactId>snmp4j-agent</artifactId>
+                <version>1.4.3</version>
+            </dependency>
```
Comment (sanjay-saxena) [inline management/pom.xml]: If there are tabs in the file, please replace them with four spaces.

File: management/src/test/java/org/kaazing/gateway/management/jmx/JmxIT.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        k3po.awaitBarrier("SESSION_ESTABLISHED");
+        String attributeNames[] = {"NumberOfCurrentSessions", "NumberOfCurrentNativeSessions", "NumberOfCurrentEmulatedSessions",
+                "TotalMessageSentCount"};
+        // Could do this cleaner
+        MBeanServerConnection mbeanServerConn = jmxConnection.getConnection();
+        Set<ObjectName> mbeanNames = mbeanServerConn.queryNames(null, null);
+        String MBeanPrefix = "subtype=services,serviceType=echo";
+        for (ObjectName name : mbeanNames) {
+            if (name.toString().indexOf(MBeanPrefix) > 0) {
+                AttributeList attrValues = mbeanServerConn.getAttributes(name, attributeNames);
```
Comment (sanjay-saxena) [inline management/src/test/java/org/kaazing/gateway/management/jmx/JmxIT.java]: If the test is **only** interested in `NumberOfCurrentSessions` attribute, then it's value can be obtained like this:

```java
     Object numberOfCurrentSessions = mbeanServerConn.getAttribute(name, "NumberOfCurrentSessions");
```

File: management/src/test/java/org/kaazing/gateway/management/jmx/JmxIT.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                                .description("realm for jmx")
+                                .httpChallengeScheme("Application Basic")
+                                .loginModule()
+                                    .type("file")
+                                    .success("required")
+                                    .option("file", "jaas-config.xml")
+                                .done()
+                            .done()
+                        .done()
+                        .property(GATEWAY_CONFIG_DIRECTORY_PROPERTY, "/Users/David/Desktop/kaazing-websocket-gateway-4.0.9/conf")
```
Comment (sanjay-saxena) [inline management/src/test/java/org/kaazing/gateway/management/jmx/JmxIT.java]: We will need to come with a way to make this work without using absolute paths.

File: management/src/test/java/org/kaazing/gateway/management/jmx/JmxRule.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                        properties.get("javax.net.ssl.trustStore"));
+            }
+
+            Map<String, Object> env = new HashMap<String, Object>();
+            SslRMIClientSocketFactory csf = new SslRMIClientSocketFactory();
+            SslRMIServerSocketFactory ssf = new SslRMIServerSocketFactory();
+            env.put(RMIConnectorServer.RMI_CLIENT_SOCKET_FACTORY_ATTRIBUTE, csf);
+            env.put(RMIConnectorServer.RMI_SERVER_SOCKET_FACTORY_ATTRIBUTE, ssf);
+            env.put(JMXConnector.CREDENTIALS, new String[]{"admin", "admin"});
+            JMXConnector jmxConnector = JMXConnectorFactory.connect(new JMXServiceURL(jmxURI), env);
```
Comment (sanjay-saxena) [inline management/src/test/java/org/kaazing/gateway/management/jmx/JmxRule.java]: Were you able to successfully connect?

File: management/pom.xml (hunk 3 above)
Comment (dpwspoon) [inline management/pom.xml]: These aren't tabs.  Thanks

File: management/src/test/java/org/kaazing/gateway/management/jmx/JmxRule.java (hunk 6 above)
Comment (dpwspoon) [inline management/src/test/java/org/kaazing/gateway/management/jmx/JmxRule.java]: Yes

File: management/src/test/java/org/kaazing/gateway/management/jmx/TlsTestUtil.java (hunk 1 above)
Comment (dpwspoon) [inline management/src/test/java/org/kaazing/gateway/management/jmx/TlsTestUtil.java]: Sorry, forgot to switch this back, fixed in next update

File: management/src/test/java/org/kaazing/gateway/management/jmx/JmxIT.java (hunk 5 above)
Comment (dpwspoon) [inline management/src/test/java/org/kaazing/gateway/management/jmx/JmxIT.java]: sorry, forgot to fix this, fixed in next commit

File: management/src/test/java/com/kaazing/gateway/management/test/util/TlsTestUtil.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+import java.io.File;
+import java.io.FileInputStream;
+import java.security.KeyStore;
+
+public class TlsTestUtil {
+    static char[] password = "ab987c".toCharArray();
+    static String keyStoreFileLocation = "target/truststore/keystore.db";
+
+    public static char[] password() {
+        return password;
```
Comment (dpwspoon) [inline management/src/test/java/com/kaazing/gateway/management/test/util/TlsTestUtil.java]: @sanjay-saxena why did you want a clone here?  Wasn't following reason, happy to make change (We will have to make the same change elsewhere)

Comment (sanjay-saxena) [inline management/src/test/java/com/kaazing/gateway/management/test/util/TlsTestUtil.java]: The method is returning a static array containing the password as-is. Ideally, you don't want anybody to modify that password as the array is mutable.

**Comment by dpwspoon** [comment]
@sanjay-saxena applied feedback

**Comment by dpwspoon** [comment]
This framework is in 4.x and will be forward ported.  That was the point of this.  Closing this for now, we can revisit these tests to see if they are needed in the future

**Pull request closed without merging**